jax/_src/numpy/lax_numpy.py

```diff
@@ -986,7 +986,12 @@ def heaviside(x1, x2):
 def hypot(x1, x2):
   _check_arraylike("hypot", x1, x2)
   x1, x2 = _promote_dtypes_inexact(x1, x2)
-  return lax.sqrt(x1*x1 + x2*x2)
+  x1 = lax.abs(x1)
+  x2 = lax.abs(x2)
+  if x2 > x1:
```

Comment:
You need to use `lax.select` or `jnp.where` here, instead of Python-level control flow; otherwise the function won't be `jit`-able.